### DISCUSSION TITLE
is it possible to have mainnet fork using hardhat node to expose for frontend / backend apps

### DISCUSSION THREAD

**Discussion by Sufiyan-dev** [NomicFoundation/hardhat#5225] [Troubleshooting] [answered]
i am working on 1inch integration into my smart contract, were they don't have any contract on testnet. So I am testing it over mainnet fork. so now I want to run my full application over mainnet fork that it works. Is that possible ?

**Comment by kanej** [comment, marked as answer]
Yes, you can fork mainnet based on this guide: https://hardhat.org/hardhat-network/docs/guides/forking-other-networks

The local hardhat node exposes an JSON-RPC endpoint that you can connect to from your frontend/backend app at `http://127.0.0.1:8545/`
